src/shader/shaderLightSampling.js

```diff
@@ -27,20 +67,50 @@ LightSampleRec lightsClosestHit( sampler2D lights, uint lightCount, vec3 rayOrig
 			continue;
 		}
 
-		u *= 1.0 / dot(u, u);
-		v *= 1.0 / dot(v, v);
+		u *= 1.0 / dot( u, u );
+		v *= 1.0 / dot( v, v );
 
 		float dist;
-		if ( intersectsRectangle( light.position, normal, u, v, rayOrigin, rayDirection, dist ) ) {
 
-			if ( dist < lightSampleRec.dist || !lightSampleRec.hit ) {
+		if( light.type == 0 ) {
```

Comment:
Can we add a comment denoting which light is associated with each of these conditions?

Reply:
Yes, I think defining them as `LIGHT_TYPE_AREA`, `LIGHT_TYPE_SPOTLIGHT` and setting these as defines would be nice. Similarly I think the same treatment for texture strides (such as `MATERIAL_PIXELS` should also be passed in this way, avoiding the need to edit it in two places).